packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -8937,32 +8937,37 @@ namespace Tpetra {
         // from EpetraExt_MMHelpers.cpp: build_type2_exports
         std::vector<bool> IsOwned(numCols,true);
         std::vector<int>  SentTo(numCols,-1);
-        if(!MyImporter.is_null())
-            for(auto && rlid : MyImporter->getRemoteLIDs() ) // the remoteLIDs must be from sourcematrix
-                IsOwned[rlid]=false;
+        if (! MyImporter.is_null ()) {
+          for (auto && rlid : MyImporter->getRemoteLIDs()) { // the remoteLIDs must be from sourcematrix
+            IsOwned[rlid]=false;
+          }
+        }
 
         std::vector<std::pair<int,GO> > usrtg;
         usrtg.reserve(TEPID2.size());
 
-        for(uint i=0;i<TEPID2.size() ;++i) {
-            const LO  row = TELID2[i];
-            const int pid = TEPID2[i];
-            for(uint j=rowptr[row]; j<rowptr[row+1]; j++) {
-                    int col=colind[j];
-                    if(IsOwned[col] && SentTo[col]!=pid){
-                        SentTo[col]    = pid;
-                        GO gid = this->getColMap()->getGlobalElement(col); // *this is sourcematrix
-                        usrtg.push_back(std::pair<int,GO>(pid,gid));
-                    }
-            }
-        }
+	{
+	  const auto& colMap = * (this->getColMap ()); // *this is sourcematrix
```

Comment:
Calling `getColMap()` invokes RCP's constructor, and thus touches the reference count.  This costs something in tight loops.